src/test/java/com/hubspot/jinjava/lib/tag/eager/EagerDoTagTest.java

```diff
@@ -0,0 +1,57 @@
+package com.hubspot.jinjava.lib.tag.eager;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.hubspot.jinjava.ExpectedNodeInterpreter;
+import com.hubspot.jinjava.JinjavaConfig;
+import com.hubspot.jinjava.interpret.DeferredValue;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.lib.tag.DoTagTest;
+import com.hubspot.jinjava.lib.tag.Tag;
+import com.hubspot.jinjava.mode.EagerExecutionMode;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class EagerDoTagTest extends DoTagTest {
+  private Tag tag;
+  private ExpectedNodeInterpreter expectedNodeInterpreter;
+
+  @Before
+  public void eagerSetup() {
+    interpreter =
+      new JinjavaInterpreter(
+        jinjava,
+        context,
+        JinjavaConfig.newBuilder().withExecutionMode(new EagerExecutionMode()).build()
+      );
+
+    tag = new EagerDoTag();
+    context.registerTag(tag);
+    context.put("deferred", DeferredValue.instance());
+    expectedNodeInterpreter =
+      new ExpectedNodeInterpreter(interpreter, tag, "tags/eager/dotag");
+    JinjavaInterpreter.pushCurrent(interpreter);
+  }
+
+  @After
+  public void teardown() {
+    JinjavaInterpreter.popCurrent();
+  }
+
+  @Test
+  public void itHandlesDeferredDo() {
+    context.put("foo", 2);
+    expectedNodeInterpreter.assertExpectedOutput("handles-deferred-do");
```

Comment:
just as a personal preference, if inputs and expected files are short enough, I would just put them right in the code here. It's way easier to read when you don't have to flip back and forth  between files.

Someday we'll get [text blocks](https://openjdk.java.net/jeps/355).

Reply:
Okay, I'll make a note to clean up the one-liners, and text blocks would be perfect here if we get them